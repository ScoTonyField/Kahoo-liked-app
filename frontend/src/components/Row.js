import React from 'react';
import PropTypes from 'prop-types';
import Box from '@material-ui/core/Box';
import Collapse from '@material-ui/core/Collapse';
import IconButton from '@material-ui/core/IconButton';
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';
import Typography from '@material-ui/core/Typography';
import KeyboardArrowDownIcon from '@material-ui/icons/KeyboardArrowDown';
import KeyboardArrowUpIcon from '@material-ui/icons/KeyboardArrowUp';
import { Button } from '@material-ui/core';

function Row (props) {
  const [open, setOpen] = React.useState(false);

  return (
    <React.Fragment>
      <TableRow>
        <TableCell>
          <IconButton aria-label="expand row" size="small" onClick={() => setOpen(!open)}>
            {open ? <KeyboardArrowUpIcon /> : <KeyboardArrowDownIcon />}
          </IconButton>
        </TableCell>
        <TableCell align="center">
          {props.row.qid}
        </TableCell>
        {/* <TableCell align="center">{props.row.isSingle}</TableCell> */}
        {
          props.row.isSingle
            ? (
              <TableCell align="center">Single-select</TableCell>
              )
            : (
              <TableCell align="center">Multiple-select</TableCell>
              )
        }
        <TableCell align="center">
          <Button variant="contained" color="primary" onClick={ (event) => { props.edit(props.row.qid, event) }}>
            Edit
          </Button>
        </TableCell>
        <TableCell align="center">
          <Button variant='contained' color='secondary' onClick={(event) => { props.remove(props.row.qid, event) }}>
            Delete
          </Button>
        </TableCell>
      </TableRow>
      <TableRow>
        <TableCell style={{ paddingBottom: 0, paddingTop: 0 }} colSpan={6}>
          <Collapse in={open} timeout="auto" unmountOnExit>
            <Box margin={1}>
              <Typography variant='h5' component='div'>
                {props.row.contents}
              </Typography>
              {props.row.options.map((data) => (
                <Typography variant='body1' component='div' key={data}>
                  {data}
                </Typography>
              ))}
            </Box>
          </Collapse>
        </TableCell>
      </TableRow>
    </React.Fragment>
  );
}

Row.propTypes = {
  row: PropTypes.shape({
    qid: PropTypes.string.isRequired,
    isSingle: PropTypes.bool.isRequired,
    contents: PropTypes.string.isRequired,
    timelimit: PropTypes.number.isRequired,
    points: PropTypes.number.isRequired,
    media: PropTypes.any,
    options: PropTypes.array.isRequired,
    answers: PropTypes.array.isRequired,
  }).isRequired,
  remove: PropTypes.func,
  edit: PropTypes.func,
};

export default Row;
