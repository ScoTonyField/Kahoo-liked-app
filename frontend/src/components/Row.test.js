import { TableCell, Typography } from '@material-ui/core';
import { shallow } from 'enzyme';
import React from 'react';
import Row from './Row';

describe('Row', () => {
  let row;
  beforeEach(() => {
    const props = {
      row: {
        qid: '12345',
        isSingle: true,
        contents: 'Hello',
        timeLimit: 10,
        points: 25,
        media: '',
        options: ['options1', 'options2'],
        answers: [0]
      }
    };
    row = shallow(<Row {...props} />);
  });

  // check snapshot
  it('snapshot', () => {
    expect(row).toMatchSnapshot();
  });

  // checked qid is correct
  it('checked qid', () => {
    expect(row.find(TableCell).at(1).text()).toEqual('12345');
  });

  // checked Question and option content is correct
  it('initial close', () => {
    expect(row.find(Typography).at(0).text()).toEqual('Questions:Hello Options: options1options2');
  });
})
