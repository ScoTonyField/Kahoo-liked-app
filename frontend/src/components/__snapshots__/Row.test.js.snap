// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row snapshot 1`] = `ShallowWrapper {}`;
